circle.yml

```diff
@@ -22,8 +22,8 @@ jobs:
     - run: go generate github.com/gopherjs/gopherjs/compiler/prelude
     - run: diff -u <(echo -n) <(git status --porcelain)
     - run: diff -u <(echo -n) <(gofmt -d .)
-    - run: go tool vet *.go # Go package in root directory.
-    - run: for d in */; do echo $d; done | grep -v tests/ | grep -v third_party/ | xargs go tool vet # All subdirectories except "tests", "third_party".
+    - run: go vet *.go # Go package in root directory.
+    - run: for d in */; do echo ./$d...; done | grep -v ./doc | grep -v ./test | grep -v ./node | xargs go vet # All subdirectories except "doc", "tests", "node*".
```

Comment:
Please explain this change.

Reply:
* `go tool vet` is deprecated and `go vet` is the substitute. ([Error](https://circleci.com/gh/gopherjs/gopherjs/1672))
* It looked like `go vet` required the relative paths for local packages. I thought this line tries to `go vet` against part of the local directories. Is that correct?

Reply:
This change makes sense to me, as @hajimehoshi described above.

A few questions/suggestions:

1. I suspect the `go tool vet *.go` line can be simplified even further to just `go vet`, or `go vet .`, to run vet on the Go package with the import path `github.com/gopherjs/gopherjs`. It was `*.go` previously because `go tool vet` did not accept import paths, only files and directories.

2. Does `doc` directory need to be skipped? It has just 2 .md files, so the import path pattern `./doc/...` should not be matching any packages.

Reply:
> 1.

Done.

> 2.

`go vet ./doc/...` fails with the exit code non-zero, and I thought I needed to avoid.

Would `go vet ./...` work btw?

Reply:
```
hajimehoshi@Hajimes-MacBook-Pro ~/go/src/github.com/gopherjs/gopherjs
$ go vet ./docs/...
go: warning: "./docs/..." matched no packages
no packages to vet

hajimehoshi@Hajimes-MacBook-Pro ~/go/src/github.com/gopherjs/gopherjs
$ echo $?
1
```

Reply:
> Would `go vet ./...` work btw?

I don't think it would, because the intention is to skip "tests" directory from being vetted.